coordinator/src/routes/dispense/verify-signature.ts

```diff
@@ -5,83 +5,108 @@ import {
+import {getRequestId} from "../../utils/headers"
+import {extractHl7v3PrescriptionFromMessage} from "../../services/communication/tracker/tracker-response-parser"
 import {isBundle} from "../../utils/type-guards"
-import {convertParentPrescription} from "../../services/translation/request/prescribe/parent-prescription"
-import {
-  verifyPrescriptionSignatureValid,
-  verifySignatureHasCorrectFormat,
-  verifySignatureDigestMatchesPrescription
-} from "../../services/signature-verification"
-import pino from "pino"
+import {getMedicationRequests} from "../../services/translation/common/getResourcesOfType"
+import {verifySignature} from "../../services/signature-verification"
 import {buildVerificationResultParameter} from "../../utils/build-verification-result-parameter"
 
+// todo:
+// 1. Remove VerifySignatureTemp payload - DONE
+// 2. Ensure endpoint accepts the following types of payload:
+//  2a. release response - DONE
+//  2b. parent prescription - DONE
+// 3. Re-instate external validator - DONE
+// 4. Extract prescription id(s)/repeat-numbers from request - DONE
+// 5. Use extracted prescription id(s) to track hl7v3 prescription(s) - DONE
+// 6. Verify digest, signature for each prescription - DONE
+// 7. Return parameters result as before - DONE
+// 8. Test cases
+//  8a. HL7v3 Acute/Repeat Prescribing/Repeat Dispensing prescriptions
+//  8b. FHIR Acute/Repeat Prescribing/Repeat Dispensing prescriptions
+//  8c. No prescription in tracker for prescription id(s) in request
+// 9. Error handling for no prescription found in tracker
+// 10. Update smoke-tests
+
 export default [
-  /*
-      Verify prescription bundle signatures.
-    */
   {
     method: "POST",
     path: `${BASE_PATH}/$verify-signature`,
     handler: externalValidator(
-      async (request: Hapi.Request, responseToolkit: Hapi.ResponseToolkit) => {
-        const outerBundle = getPayload(request) as fhir.Resource
-        if (!isBundle(outerBundle)) {
+      async (request: Hapi.Request, responseToolkit: Hapi.ResponseToolkit): Promise<Hapi.ResponseObject> => {
+        const messageId = getRequestId(request.headers)
+
+        const bundle = getPayload(request) as fhir.Resource
+        if (!isBundle(bundle)) {
           const operationOutcome = fhir.createOperationOutcome([
             errors.createResourceTypeIssue("Bundle")
           ])
           return responseToolkit.response(operationOutcome).code(400).type(ContentTypes.FHIR)
         }
 
-        request.logger.info("Verifying prescription signatures from Bundle")
-        const verificationResponses = outerBundle.entry
-          .map(entry => entry.resource)
-          .filter(isBundle)
-          .map((innerBundle, index) => verifyPrescriptionSignature(innerBundle, index, request.logger))
+        const prescriptions = bundle.entry.some(entry => isBundle(entry.resource))
+          ? bundle.entry
+              .map(entry => entry.resource)
+              .filter(isBundle)
+          : [bundle]
```

Comment:
add some appropriately named functions to make it clear this is handling release-responses or fhir prescription